Unit_testing_frameworks/Calculator_Tests/CosTests.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using CSharpCalculator;
+
+namespace Calculator_Tests
+{
+    [TestClass]
+	public class CosTests
+	{
+        [TestMethod]
+        [ExpectedException(typeof(NotFiniteNumberException))]
+        public void CosWrongInputMassage()
```

Comment:
Why are you doing so many checks in the method? Just make the same method as you did in AbsTests class. [ExpectedException(typeof(NotFiniteNumberException))] attribute is enough for test. And "Assert.AreEqual("Wrong input", ex.Message);" doesn't work here because of the exception on line 18.

Reply:
Fixed